src/merkle_reg.rs

```diff
@@ -11,7 +11,7 @@ use crate::traits::{CmRDT, CvRDT};
 pub type Hash = [u8; 32];
 
 /// A node in the Merkle DAG
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, PartialOrd, Deserialize)]
```

Comment:
If you're going to derive PartialOrd, might as well get Ord as well. Also keep the serde(or any third party) derives together at the end.
```suggestion
#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
```